### PR TITLE
Better orElseSucceed method (#375)

### PR DESCRIPTION
**Pull request by bmarinovic**

### CODE REVIEW

**Comment by sideeffffect** [comment]
Nice. The only thing remaining is to make Dotty with the code. Try this locally on your computer by changing `scalaVersion in ThisBuild` to `ScalaDotty` in `project/BuildHelper.scala` :wink:

**Comment by sideeffffect** [comment]
@bmarinovic could you please fix the merge conflicts? :pray:

**Comment by bmarinovic** [comment]
I will on a next commit when we find a better way to type inference to work on Dotty. Currently, it doesn't make sense to merge this.

**Comment by sideeffffect** [comment]
Any luck moving this forward? :smiley:

**Comment by bmarinovic** [comment]
It is stuck at Dotty's type inference problems.
Until we find concise way to do it, I believe no.
@adamgfraser can double check it

File: src/main/scala/zio/prelude/fx/ZPure.scala [hunk 1]
```diff
@@ -460,6 +446,18 @@ sealed trait ZPure[-S1, +S2, -R, +E, +A] { self =>
...
 }
 
+trait ZPureSyntax {
+  final implicit class ZPureOps[S1, S2, R, E, A](private val self: ZPure[S1, S2, R, E, A]) {
+
+    /**
+     * Executes this computation and returns its value, if it succeeds, but
+     * otherwise succeeds with the specified value.
+     */
+    def orElseSucceed[A1 >: A](a1: => A1)(implicit ev: S1 <:< S2): ZPure[S1, S2, R, Nothing, A1] =
```
Comment (jdegoes) [inline src/main/scala/zio/prelude/fx/ZPure.scala:456]: If you are going to require implicit evidence here that `S1 <:< S2`, then you may as well just have one type parameter `S` on the `ZPureOps`. It's the same thing only will infer better.

Comment (adamgfraser) [reply src/main/scala/zio/prelude/fx/ZPure.scala:456]: Yes. That is the approach I took with `UnifiedSyntax` in #461.

**Comment by jdegoes** [comment]
@bmarinovic I will close this for now. If you can fix conflicts and simplify the encoding a bit (which should get dotty compiling too, hopefully), then we can merge it in right away. Thanks for your work on this!

**Pull request closed without merging**